Converting instances to 0.7.0 fails on some instances

## Problem

After running the conversion script, some instances fail to load with the following message:

```
[ERROR] InstanceService - Failed to load the instances: play.api.libs.json.JsResultException: JsResultException(errors:List((/template/parameterInfos,List(ValidationError(List(error.expected.jsobject),WrappedArray())))))
```

Turned out that those instances didn't have any parameterInfos before:

```
"parameterInfos": {}
```

The script converted that to:

```
"parameterInfos": null
```

Looks like a version mismatch of JQ...

```
$ jq --version
jq version 1.3
$ cat test-sftp.json | jq '.template.parameterInfos = (.template.parameterInfos | with_entries(.value.id = .value.name))' | jq .template.parameterInfos
null
```

```
$ jq --version
jq-1.5
$ cat test-sftp.json | jq '.template.parameterInfos = (.template.parameterInfos | with_entries(.value.id = .value.name))' | jq .template.parameterInfos
{}
```

--- a/script/instances-0.6.0-to-0.7.0.sh
+++ b/script/instances-0.6.0-to-0.7.0.sh
@@ -8,19 +8,9 @@ fi
 set -euo pipefail
 
 jq_version="$(jq --version 2>&1)" # redirecting stderr https://github.com/stedolan/jq/issues/1452
-continue=0
 if [[ $jq_version != *"1.5"* ]]; then
-  echo "WARNING: This script was tested against jq-1.5 and should be run against that as well."
-  read -p "WARNING: Looks like you are using $jq_version. Continue anyway? [y/n] " -n 1 -r
-  echo
-  if [[ $REPLY =~ ^[Yy]$ ]]; then
-    continue=1
-  fi
+  echo "ERROR: This script was tested against jq-1.5 and should be run against that as well."
 else
-  continue=1
-fi
-
-if [[ "$continue" == 1 ]]; then
   instanceDir="$1"
   backupDir="$instanceDir.bak_$(date +%s)"
 

--- a/script/instances-0.6.0-to-0.7.0.sh
+++ b/script/instances-0.6.0-to-0.7.0.sh
@@ -7,25 +7,40 @@ fi
 
 set -euo pipefail
 
-instanceDir="$1"
-backupDir="$instanceDir.bak_$(date +%s)"
-
-echo "Backing up $instanceDir to $backupDir."
-cp -r $instanceDir $backupDir
-echo "Converting instances format in $instanceDir from Broccoli <0.6.0 to 0.7.0."
-for instanceFile in $instanceDir/*.json; do
-  echo "- Converting $instanceFile"
-  tmpInstanceFile="$instanceFile.tmp"
-  cat $instanceFile | jq '.template.parameterInfos = (.template.parameterInfos | with_entries(.value.id = .value.name))' > $tmpInstanceFile
-  mv $tmpInstanceFile $instanceFile
-done
-
-echo "Conversion finished. Looks like everything went well."
-read -p "Delete $backupDir? [y/n] " -n 1 -r
-echo
-if [[ $REPLY =~ ^[Yy]$ ]]; then
-  echo "Deleting $backupDir."
-  rm -rf $backupDir
+jq_version="$(jq --version 2>&1)" # redirecting stderr https://github.com/stedolan/jq/issues/1452
+continue=0
+if [[ $jq_version != *"1.5"* ]]; then
+  echo "WARNING: This script was tested against jq-1.5 and should be run against that as well."
+  read -p "WARNING: Looks like you are using $jq_version. Continue anyway? [y/n] " -n 1 -r
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    continue=1
+  fi
 else
-  echo "Keeping $backupDir for now. You have to delete it manually."
+  continue=1
+fi
+
+if [[ "$continue" == 1 ]]; then
+  instanceDir="$1"
+  backupDir="$instanceDir.bak_$(date +%s)"
+
+  echo "Backing up $instanceDir to $backupDir."
+  cp -r $instanceDir $backupDir
+  echo "Converting instances format in $instanceDir from Broccoli <0.6.0 to 0.7.0."
+  for instanceFile in $instanceDir/*.json; do
+    echo "- Converting $instanceFile"
+    tmpInstanceFile="$instanceFile.tmp"
+    cat $instanceFile | jq '.template.parameterInfos = (.template.parameterInfos | with_entries(.value.id = .value.name))' > $tmpInstanceFile
+    mv $tmpInstanceFile $instanceFile
+  done
+
+  echo "Conversion finished. Looks like everything went well."
+  read -p "Delete $backupDir? [y/n] " -n 1 -r
+  echo
+  if [[ $REPLY =~ ^[Yy]$ ]]; then
+    echo "Deleting $backupDir."
+    rm -rf $backupDir
+  else
+    echo "Keeping $backupDir for now. You have to delete it manually."
+  fi
 fi

--- a/script/instances-0.6.0-to-0.7.0.sh
+++ b/script/instances-0.6.0-to-0.7.0.sh
@@ -20,7 +20,7 @@ else
   for instanceFile in "$instanceDir/*.json"; do
     echo "- Converting $instanceFile"
     tmpInstanceFile="$instanceFile.tmp"
-    cat "$instanceFile" | jq '.template.parameterInfos = (.template.parameterInfos | with_entries(.value.id = .value.name))' > "$tmpInstanceFile"
+    jq '.template.parameterInfos = (.template.parameterInfos | with_entries(.value.id = .value.name))' < "$instanceFile" > "$tmpInstanceFile"
     mv "$tmpInstanceFile" "$instanceFile"
   done
 

--- a/script/instances-0.6.0-to-0.7.0.sh
+++ b/script/instances-0.6.0-to-0.7.0.sh
@@ -19,7 +19,8 @@ else
   echo "Converting instances format in $instanceDir from Broccoli <0.6.0 to 0.7.0."
   for instanceFile in "$instanceDir"/*.json; do
     echo "- Converting $instanceFile"
-    tmpInstanceFile="$(mktemp -t $(basename $instanceFile))"
+    instanceFileName=$(basename "$instanceFile")
+    tmpInstanceFile=$(mktemp -t "$instanceFileName")
     jq '.template.parameterInfos = (.template.parameterInfos | with_entries(.value.id = .value.name))' < "$instanceFile" > "$tmpInstanceFile"
     mv "$tmpInstanceFile" "$instanceFile"
   done

--- a/script/instances-0.6.0-to-0.7.0.sh
+++ b/script/instances-0.6.0-to-0.7.0.sh
@@ -15,13 +15,13 @@ else
   backupDir="$instanceDir.bak_$(date +%s)"
 
   echo "Backing up $instanceDir to $backupDir."
-  cp -r $instanceDir $backupDir
+  cp -r "$instanceDir" "$backupDir"
   echo "Converting instances format in $instanceDir from Broccoli <0.6.0 to 0.7.0."
-  for instanceFile in $instanceDir/*.json; do
+  for instanceFile in "$instanceDir/*.json"; do
     echo "- Converting $instanceFile"
     tmpInstanceFile="$instanceFile.tmp"
-    cat $instanceFile | jq '.template.parameterInfos = (.template.parameterInfos | with_entries(.value.id = .value.name))' > $tmpInstanceFile
-    mv $tmpInstanceFile $instanceFile
+    cat "$instanceFile" | jq '.template.parameterInfos = (.template.parameterInfos | with_entries(.value.id = .value.name))' > "$tmpInstanceFile"
+    mv "$tmpInstanceFile" "$instanceFile"
   done
 
   echo "Conversion finished. Looks like everything went well."
@@ -29,7 +29,7 @@ else
   echo
   if [[ $REPLY =~ ^[Yy]$ ]]; then
     echo "Deleting $backupDir."
-    rm -rf $backupDir
+    rm -rf "$backupDir"
   else
     echo "Keeping $backupDir for now. You have to delete it manually."
   fi

--- a/script/instances-0.6.0-to-0.7.0.sh
+++ b/script/instances-0.6.0-to-0.7.0.sh
@@ -17,7 +17,7 @@ else
   echo "Backing up $instanceDir to $backupDir."
   cp -r "$instanceDir" "$backupDir"
   echo "Converting instances format in $instanceDir from Broccoli <0.6.0 to 0.7.0."
-  for instanceFile in "$instanceDir/*.json"; do
+  for instanceFile in "$instanceDir"/*.json; do
     echo "- Converting $instanceFile"
     tmpInstanceFile="$instanceFile.tmp"
     jq '.template.parameterInfos = (.template.parameterInfos | with_entries(.value.id = .value.name))' < "$instanceFile" > "$tmpInstanceFile"

--- a/script/instances-0.6.0-to-0.7.0.sh
+++ b/script/instances-0.6.0-to-0.7.0.sh
@@ -19,7 +19,7 @@ else
   echo "Converting instances format in $instanceDir from Broccoli <0.6.0 to 0.7.0."
   for instanceFile in "$instanceDir"/*.json; do
     echo "- Converting $instanceFile"
-    tmpInstanceFile="$instanceFile.tmp"
+    tmpInstanceFile="$(mktemp -t $(basename $instanceFile))"
     jq '.template.parameterInfos = (.template.parameterInfos | with_entries(.value.id = .value.name))' < "$instanceFile" > "$tmpInstanceFile"
     mv "$tmpInstanceFile" "$instanceFile"
   done

